app/editor/src/store/store.ts

```diff
@@ -5,6 +5,7 @@ import logger from 'redux-logger';
 
 import { reducer } from './rootReducer';
 
+export type ActionDelegate<T> = T | ((values: T) => T);
```

Comment:
Very cool delegate to allow us to pass state or a function that receives the current state before returning new state.